sdk/iot/Azure.Iot.Hub.Service/src/Authentication/ISasTokenProvider.cs

```diff
@@ -0,0 +1,17 @@
+﻿// Copyright (c) Microsoft Corporation. All rights reserved.
+// Licensed under the MIT License.
+
+namespace Azure.Iot.Hub.Service.Authentication
+{
+    /// <summary>
+    /// The token provider interface for shared access signature based authentication.
+    /// </summary>
+    public interface ISasTokenProvider
+    {
+        /// <summary>
+        /// Retrieve the shared access signature to be used.
+        /// </summary>
+        /// <returns>The shared access signature to be used for authenticating HTTP requests to the service.</returns>
```

Comment:
Worth adding a note that this function is called once per http request. That there is no caching and re-using